src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureKeyVaultReference/AzureKeyVaultSecretProvider.cs

```diff
@@ -16,12 +16,15 @@ internal class AzureKeyVaultSecretProvider
         private readonly IDictionary<string, SecretClient> _secretClients;
         private readonly TokenCredential _credential;
         private readonly Func<Uri, ValueTask<string>> _secretResolver;
+        private readonly Dictionary<string, TimeSpan> _secretRefreshIntervals = new Dictionary<string, TimeSpan>();
+        private HashSet<CachedKeyVaultSecret> _cachedKeyVaultSecrets = new HashSet<CachedKeyVaultSecret>();
```

Comment:
>_cachedKeyVaultSecrets [](start = 46, length = 22)

Is this class used in any multi-threading?

Reply:
Switched to using a thread safe collection for cached key vault secrets.